backend/app/core/validation/util.py

```diff
@@ -0,0 +1,57 @@
+import logging
+from typing import Any, Collection, Mapping, Optional, Sequence
+
+from sqlalchemy.orm import Session
+
+from app.api.api_v1.routers.lookups.util import get_metadata
+
+_LOGGER = logging.getLogger(__file__)
+
+
+def _flatten_maybe_tree(
+    maybe_tree: Sequence[Mapping[str, Any]],
+    values: Optional[set[str]] = None,
+) -> Collection[str]:
```

Comment:
I know you said you would ... but putting a note here for unit tests :D

Reply:
done